unit/java_bytecode/java_bytecode_parse_generics/parse_generic_class_with_generic_inner_classes.cpp

```diff
@@ -0,0 +1,1446 @@
+/*******************************************************************\
+
+ Module: Unit tests for parsing generic classes
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <testing-utils/require_symbol.h>
+#include <testing-utils/require_type.h>
+
+#include <util/cmdline.h>
+#include <util/config.h>
+#include <util/language.h>
+#include <util/prefix.h>
+#include <util/std_types.h>
+
+#include <java_bytecode/java_bytecode_language.h>
+
+#include <iostream>
+#include <testing-utils/load_java_class.h>
+
+SCENARIO(
+  "parse_generic_class_with_generic_inner_classes_fields",
+  "[core][java_bytecode][java_bytecode_parse_generics]")
+{
+  const symbol_tablet &new_symbol_table = load_java_class(
+    "GenericClassWithGenericInnerClasses",
+    "./java_bytecode/java_bytecode_parse_generics");
+
+  std::string class_prefix = "java::GenericClassWithGenericInnerClasses";
+  THEN("There should be a symbol for "
+    "GenericClassWithGenericInnerClasses with correct components")
+  {
+    REQUIRE(new_symbol_table.has_symbol(class_prefix));
+    const symbolt &class_symbol = new_symbol_table.lookup_ref(class_prefix);
+
+    const class_typet &class_type =
+      require_symbol::require_complete_class(class_symbol);
+
+    THEN("The field component should be a pointer to "
+      "GenericClassWithGenericInnerClasses$InnerClass")
+    {
+      const struct_typet::componentt &field_component =
+        require_type::require_component(class_type, "field");
+
+      require_type::require_pointer(
+        field_component.type(), symbol_typet
+          (class_prefix + "$InnerClass"));
+
+      THEN("The pointer should be generic")
+      {
+        REQUIRE(is_java_generic_type(field_component.type()));
+        const auto &generic_variables =
+          to_java_generic_type(field_component.type()).generic_type_variables();
+        REQUIRE(generic_variables.size() == 1);
+        const java_generic_parametert &generic_param = generic_variables[0];
+        REQUIRE(
+          generic_param.type_variable() ==
+          symbol_typet(class_prefix + "::T"));
```

Comment:
Nit: I'm not sure if this is the new clang-format style or not (apologies for the noise if it is) - but indenting this line a bit would make it easier to see that it is the RHS of the `==` operator. If this is the clang-format style, then just ignore me :-)